docling_core/types/doc/document.py

```diff
@@ -1297,17 +1581,241 @@ def iterate_items(
+                if isinstance(item, PictureItem):
+
+                    if item.image is not None:
+                        img = item.image.pil_image
+
+                        hexhash = item._image_to_hexhash()
+
+                        loc_path = image_dir / f"image_{img_count:06}.png"
+                        if hexhash is not None:
+                            loc_path = image_dir / f"image_{img_count:06}_{hexhash}.png"
+
+                        abs_path = Path(loc_path).resolve()
+
+                        img.save(abs_path)
+                        uri = f"file://{abs_path}"
+
+                        item.image.uri = AnyUrl(uri)
+
+                        # if item.image._pil is not None:
+                        #    item.image._pil.close()
+
+                    img_count += 1
+
+        return result
+
     def print_element_tree(self):
-        """print_element_tree."""
+        """Print_element_tree."""
         for ix, (item, level) in enumerate(self.iterate_items(with_groups=True)):
             if isinstance(item, GroupItem):
                 print(" " * level, f"{ix}: {item.label.value} with name={item.name}")
             elif isinstance(item, DocItem):
                 print(" " * level, f"{ix}: {item.label.value}")
 
-    def export_to_dict(self) -> Dict:
-        """export_to_dict."""
-        return self.model_dump(mode="json", by_alias=True, exclude_none=True)
+    def export_to_element_tree(self) -> str:
+        """Export_to_element_tree."""
+        texts = []
+        for ix, (item, level) in enumerate(self.iterate_items(with_groups=True)):
+            if isinstance(item, GroupItem):
+                texts.append(
+                    " " * level + f"{ix}: {item.label.value} with name={item.name}"
+                )
+            elif isinstance(item, DocItem):
+                texts.append(" " * level + f"{ix}: {item.label.value}")
+
+        return "\n".join(texts)
+
+    def save_as_json(
+        self,
+        filename: Path,
+        artifacts_dir: Optional[Path] = None,
+        image_mode: ImageRefMode = ImageRefMode.EMBEDDED,
+        by_alias: bool = True,
+        exclude_none: bool = True,
+        indent: int = 2,
+    ):
+        """Save as json."""
+        if artifacts_dir is None:
+            # Remove the extension and add '_pictures'
+            artifacts_dir = filename.with_suffix("")
+            artifacts_dir = artifacts_dir.with_name(artifacts_dir.stem + "_artifacts")
+
+            os.makedirs(artifacts_dir, exist_ok=True)
+
+        new_doc: "DoclingDocument" = DoclingDocument(name=self.name)
+        if image_mode == ImageRefMode.PLACEHOLDER:
+            new_doc = self
+        elif image_mode == ImageRefMode.REFERENCED:
+            new_doc = self._with_pictures_refs(image_dir=artifacts_dir)
+        elif image_mode == ImageRefMode.EMBEDDED:
+            new_doc = self._with_embedded_pictures()
+        else:
+            raise ValueError("Unsupported ImageRefMode")
```

Comment:
it would be better to factor out this block which we are now repeating >3 times

Reply:
Factored it out now.